app/services/optout/OptOutService.scala

```diff
@@ -50,40 +51,66 @@ class OptOutService @Inject()(itsaStatusUpdateConnector: ITSAStatusUpdateConnect
     val previousYear = currentYear.previousYear
     val nextYear = currentYear.nextYear
 
+    fetchOptOutInitialState(previousYear, currentYear, nextYear).
+      map(initialState => {
+      createOptOutProposition(previousYear, currentYear, nextYear, initialState)
+    })
+  }
+
+  private def fetchOptOutInitialState(previousYear: TaxYear,
+                                      currentYear: TaxYear,
+                                      nextYear: TaxYear)
+                                     (implicit user: MtdItUser[_],
+                                      hc: HeaderCarrier,
+                                      ec: ExecutionContext): Future[OptOutInitialState] = {
+
     val finalisedStatusFuture: Future[Boolean] = calculationListService.isTaxYearCrystallised(previousYear)
     val statusMapFuture: Future[Map[TaxYear, ITSAStatus]] = getITSAStatusesFrom(previousYear)
 
     for {
       finalisedStatus <- finalisedStatusFuture
       statusMap <- statusMapFuture
     }
-    yield createOptOutProposition(previousYear, currentYear, nextYear, finalisedStatus, statusMap)
+    yield OptOutInitialState(finalisedStatus, statusMap(previousYear), statusMap(currentYear), statusMap(nextYear))
   }
 
-  private def getITSAStatus(year: TaxYear, statusMap: Map[TaxYear, ITSAStatus]): ITSAStatus = {
-    statusMap.getOrElse(year, ITSAStatus.NoStatus)
+  def recallOptOutProposition()(implicit hc: HeaderCarrier,
+                                ec: ExecutionContext): Future[OptOutProposition] = {
+
+    val currentYear = dateService.getCurrentTaxYear
+    val previousYear = currentYear.previousYear
+    val nextYear = currentYear.nextYear
+
+    OptionT(recallOptOutInitialState()).
+      map(initialState => {
+        createOptOutProposition(previousYear, currentYear, nextYear, initialState)
+      }).getOrElseF(Future.failed(new RuntimeException("Failed to recall Opt Out journey initial state")))
```

Comment:
also this reads better for me
![Screenshot 2024-07-12 at 15 04 57](https://github.com/user-attachments/assets/72425ed4-e98e-4648-8a8f-4959cdfd29c1)